Dockerfile.multi-stage

```diff
@@ -1,9 +1,14 @@
 FROM golang:1.13.6-alpine3.11 as builder
 
-ADD . $GOPATH/src/github.com/thanos-io/thanos
 WORKDIR $GOPATH/src/github.com/thanos-io/thanos
-
+# Change in the docker context invalidates the cache so to leverage docker
+# layer caching, moving update and installing apk packages above COPY cmd
+# More info https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#leverage-build-cache 
 RUN apk update && apk upgrade && apk add --no-cache alpine-sdk
+# Replaced ADD with COPY as add is generally to download content form link or tar files
+# while COPY supports the basic copying of local files into the container.
+# https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#add-or-copy
+COPY . .
```

Comment:
I presume we use `.` here because of `WORKDIR` above? Maybe it would be worth still keeping the whole path explicitly to make it more clear?

Reply:
Yes, because we have already added the `WORKDIR` above. Sure, I will explicitly add it to be more clear.